doc/scripts/check_links.py

```diff
@@ -0,0 +1,329 @@
+### This script reads links from a list of link files and checks if all links
```

Comment:
I'm not convinced this is necessary. There is already a linkcheck builder in sphinx-build (i.e. you can run `sphinx-build -b linkcheck ...`).

Why add this code?

Reply:
Because I didn't know about the linkcheck. ;-)

The linkcheck seems to have problems with intersphinx though, but we might be able to ignore these links. And we'll get lots of duplicates, which don't really make sense in our setup. But I guess we can live with that.

I'll see if I can get that working in our setup ...